core/src/test/java/com/coderanch/blackjack/HandTest.java

```diff
@@ -0,0 +1,200 @@
+                    new Card(Rank.KING, Suit.CLUBS)),
+                23
+            ),
+
+            new HandTestArgument(
+                List.of(new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.NINE, Suit.CLUBS)),
+                21
+            )
+        );
+
+    /**
+     * Cards to test.
+     */
+    @DataPoints
+    public static final Set<Card> CARDS = Cards.getStandardDeck();
+
+
+    /**
+     * Tests that {@link Hand#withAdditionalCard(Card)} maintains the correct score.
+     *
+     * @param handTest contains the test hand, and the target score.
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void addCard_maintainsCorrectScore(HandTestArgument handTest) {
+        assertThat(
+            "The best score must match the target",
+            handTest.getHand().bestScore(),
+            is(handTest.getTargetScore())
+        );
+    }
+
+    /**
+     * Tests that passing {@code null} for the first {@code card}
+     * when constructing a new hand causes an exception to be thrown.
+     *
+     * @param card the card to construct the hand with.
+     */
+    @Theory(nullsAccepted = false)
+    @SuppressWarnings("checkstyle:methodname")
+    public void newHand_withNullFirstCard_throwsException(Card card) {
+        assertThrows("Hand must throw exception", IllegalArgumentException.class, () -> {
+            new Hand(null, card);
+        });
+    }
+
+    /**
+     * Tests that passing {@code null} for the second {@code card}
+     * when constructing a new hand causes an exception to be thrown.
+     *
+     * @param card the card to construct the hand with.
+     */
+    @Theory(nullsAccepted = false)
+    @SuppressWarnings("checkstyle:methodname")
+    public void newHand_withNullSecondCard_throwsException(Card card) {
+        assertThrows("Hand must throw exception", IllegalArgumentException.class, () -> {
+            new Hand(card, null);
+        });
+    }
+
+    /**
+     * Tests that passing {@code null} for {@code card} when adding a new card causes an exception to be thrown.
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void addCard_withNullCard_throwsException() {
+        var hand = new Hand(new Card(Rank.ACE, Suit.CLUBS), new Card(Rank.ACE, Suit.CLUBS));
+        assertThrows("Hand must throw exception", IllegalArgumentException.class, () -> {
+            hand.withAdditionalCard(null);
+        });
+    }
+
+
+    /**
+     * Tests that hands that {@link Hand#isBust()} returns the correct answer.
+     *
+     * @param handTest contains the test hand, and the target score.
+     */
+    @Theory
+    public void isBust(HandTestArgument handTest) {
+        assertThat(
+            "The hands over the legal limit must be bust.",
+            handTest.getHand().isBust(),
+            is(equalTo(handTest.getHand().bestScore() > Hand.MAX_LEGAL_SCORE))
```

Comment:
I'd prefer it that you create a data point that consists of various hands that you know are bust, and then test that for these hands `isBust()` returns true.

Add another test that does the same thing with hands that you know ahead of time are not bust.